### PR TITLE
OCD-4264 - Create listing details download files in JSON format

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: src/app/pages/resources/download/download.jsx [hunk 1]
```diff
@@ -231,14 +287,15 @@ function ChplResourcesDownload() {
               </Button>
               <Button
                 fullWidth
-                color="secondary"
-                variant="contained"
+                color="primary"
+                variant="text"
                 id="download-chpl-definition-button"
+                disabled={definitionDisabled}
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx]: Maybe?
```suggestion
                disabled={data[selectedOption].definition === ''}
```

**Review by kekey1 — DISMISSED** [review]

File: src/app/pages/resources/download/download.jsx [hunk 2]
```diff
@@ -120,102 +148,130 @@ function ChplResourcesDownload() {
...
-                    {' '}
-                    Collection of surveillance activities that resulted in a non-conformity. This is a subset of the data available in the above &quot;Surveillance Activity&quot; file. Available as a CSV file.
+                    The 2014 Edition Products file and the 2011 Edition Products file are updated quarterly.
+                  </li>
+                </ul>
+              </li>
+              <Box className={classes.warningBox}>
+                <ReportProblemOutlinedIcon />
+                <Typography>
+                  XML Files are now something text
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx:192]: Leaving a comment so we remember to change this before merging

File: src/app/pages/resources/download/download.jsx [hunk 3]
```diff
@@ -120,102 +148,130 @@ function ChplResourcesDownload() {
...
+                Entire collection of surveillance activity reported to the CHPL.
+              </li>
+              { hasAnyRole(['ROLE_ADMIN', 'ROLE_ONC'])
+                    && (
+                      <li>
+                        <Typography gutterBottom><strong>Surveillance (Basic) (CSV):</strong></Typography>
+                        {' '}
+                        Entire collection of surveillance activity reported to the CHPL, with only basic details about non-conformities. Includes statistics on timeframes related to discovered non-conformities.
+                      </li>
+                    )}
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx]: Matt, the indentation looks weird here

File: src/app/pages/resources/download/download.jsx [hunk 4]
```diff
@@ -92,7 +119,8 @@ function ChplResourcesDownload() {
       }
       return true;
     }));
-  }, [API, getApiKey, getToken, hasAnyRole]);
+    setDefinitionDisabled(data[selectedOption].definition === '');
+  }, [API, getApiKey, getToken, hasAnyRole, selectedOption]);
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx]: If we do need to keep this in a useEffect, it should be in a different one. This useEffect is basically setting up stuff from the data passed in to the component. The one where the option changes should be a different one, as it's "on the page interactivity". I can show you what that could look like if desired

File: src/app/pages/resources/download/download.jsx [hunk 5]
```diff
@@ -84,6 +108,9 @@ function ChplResourcesDownload() {
       'Surveillance Activity': { data: `${API}/surveillance/download?api_key=${getApiKey()}&type=all`, definition: `${API}/surveillance/download?api_key=${getApiKey()}&type=all&definition=true`, label: 'Surveillance' },
       'Surveillance Non-Conformities': { data: `${API}/surveillance/download?api_key=${getApiKey()}`, definition: `${API}/surveillance/download?api_key=${getApiKey()}&definition=true`, label: 'Surveillance Non-Conformities' },
       'SVAP Summary (csv)': { data: `${API}/svap/download?api_key=${getApiKey()}`, definition: `${API}/svap/download?api_key=${getApiKey()}&definition=true`, label: 'SVAP Summary' },
+      '2015 edition products (json)': { data: `${API}/download?api_key=${getApiKey()}&edition=2015&format=json`, definition: '', label: '2015 JSON' },
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx]: Can these be moved up to the top of the object? To match the ordering they'll have in the dropdown?

File: src/app/pages/resources/download/download.jsx [hunk 6]
```diff
@@ -31,22 +35,41 @@ const useStyles = makeStyles({
...
     '& li': {
-      marginBottom: '.35em',
       lineHeight: '1.3em',
+      marginBottom: '.7em',
+      marginTop: '.7em',
     },
   },
+  warningBox: {
+    padding: '16px',
+    backgroundColor: '#fdfde7',
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx]: Matt, can we use a `palette.something` here? And on the next line?

File: src/app/pages/resources/download/download.jsx [hunk 7]
```diff
@@ -1,16 +1,20 @@
...
   CardActions,
+  Link,
   Divider,
   MenuItem,
   Typography,
   makeStyles,
 } from '@material-ui/core';
 import GetAppIcon from '@material-ui/icons/GetApp';
+import ReportProblemOutlinedIcon from '@material-ui/icons/ReportProblemOutlined';
+import CodeIcon from '@material-ui/icons/Code';
```
Comment (andlar) [inline src/app/pages/resources/download/download.jsx]: We usually alphabetize these
```suggestion
import CodeIcon from '@material-ui/icons/Code';
import GetAppIcon from '@material-ui/icons/GetApp';
import ReportProblemOutlinedIcon from '@material-ui/icons/ReportProblemOutlined';
```

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request closed without merging**